### ISSUE TITLE
InMemoryBus: A suitable constructor for type 'CQELight.Buses.InMemory.Commands.InMemoryCommandBus' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.

### ISSUE THREAD

**Issue by nozzle-1** [hybrid-technologies-solutions/CQELight#487]
With Microsoft.DependencyInjection container in ASP.NET Core 3.1 application, an exceptions throws when I try to dispatch command with IDispatcher from an endpoint of a controller.
I don't found more information in CQELight's documentation with Microsoft.DependencyInjection container.

Exception :
```bash
fail: CQELight.Dispatcher.BaseDispatcher[0]
      Dispatcher.DispatchCommandAsync() : Exception when sending command SocialNetwork.Profile.Application.Commands.SayHello on bus CQELight.Buses.InMemory.Commands.InMemoryCommandBus
      System.InvalidOperationException: A suitable constructor for type 'CQELight.Buses.InMemory.Commands.InMemoryCommandBus' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.<>c__DisplayClass7_0.<GetCallSite>b__0(Type type)
         at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.CreateServiceAccessor(Type serviceType)
         at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
         at CQELight.IoC.Microsoft.Extensions.DependencyInjection.MicrosoftScope.Resolve(Type type, IResolverParameter[] parameters)
         at CQELight.Dispatcher.BaseDispatcher.DispatchCommandAsync(ICommand command, ICommandContext context, String callerMemberName)
```

Source code:
```csharp
public class Program
{
    [...]

    public static IHostBuilder CreateHostBuilder(string[] args) =>
        Host.CreateDefaultBuilder(args)
        .ConfigureCQELight((x, config) =>
        {
            x.UseInMemoryCommandBus();
            x.UseInMemoryEventBus();
        })
        .ConfigureWebHostDefaults(webBuilder =>
        {
            webBuilder.UseStartup<Startup>();
        });
}


[ApiController]
[Route("account")]
public class AccountController : ControllerBase
{
     private readonly IDispatcher _dispatcher;
     public AccountController(IDispatcher dispatcher)
     {
         _dispatcher = dispatcher;
     }
     [HttpGet]
     public async Task<IActionResult> Get()
     {
         await _dispatcher.DispatchCommandAsync(new SayHello());
         return Ok();
     }
}

public class SayHello : ICommand
{ }

public class SayHelloConsole : ICommandHandler<SayHello>
{
    public Task<Result> HandleAsync(SayHello command, ICommandContext context = null)
    {
        Console.WriteLine("Hello world!");
        return Result.Ok();
    }
}
```

**Comment by hybrid-technologies-solutions** [comment]
Hello @nozzle-1 ,

Thanks for reporting issue.
Indeed, this is a limitation of Microsoft dependency injection system, because in memory buses doesn't have public ctor parameterless.
We'll look at it.

As a workaround, you can switch to Autofac IoC package that does the same job under the hood.

We'll update this thread when a fix will be available

**Comment by hybrid-technologies-solutions** [comment]
Bug has been fixed and wil be in next release (1.2.1)
